packages/mobile/src/components/edit/VisibilityField/VisibilityScreen.tsx

```diff
@@ -25,14 +25,26 @@ export const VisibilityScreen = () => {
   const { isEnabled: isEditableAccessEnabled } = useFeatureFlag(
     FeatureFlags.EDITABLE_ACCESS_ENABLED
   )
+  const { isEnabled: isPaidScheduledEnabled } = useFeatureFlag(
+    FeatureFlags.PAID_SCHEDULED
+  )
   const { values, initialValues, setValues } = useFormikContext<FormValues>()
-  const { is_unlisted, is_scheduled_release, release_date, isUpload } = values
-  const isAlreadyPublic = !isUpload && !initialValues.is_unlisted
+  const { entityType } = values
+  const hiddenKey = entityType === 'track' ? 'is_unlisted' : 'is_private'
+  const {
+    [hiddenKey]: isHidden,
+    is_scheduled_release,
+    release_date,
+    isUpload
+  } = values
+
+  const initiallyPublic = !isUpload && !initialValues[hiddenKey]
+  const hidePaidScheduled = !isPaidScheduledEnabled && entityType === 'album'
```

Comment:
should this not apply regardless of entity type?

I think the flag is meant to cover hidden paid tracks as well. That was also net new in this project

Reply:
oh yeah youre right, updating